subsys/bluetooth/controller/hci/hci.c

```diff
@@ -5253,6 +5271,91 @@ static void le_ext_adv_report(struct pdu_data *pdu_data,
 	adv_info->length = data_len;
 	memcpy(&adv_info->data[0], data, data_len);
 
+	if (!scan_data) {
+		node_rx_extra_list_release(node_rx->hdr.rx_ftr.extra);
+
+		return;
+	}
+
+	/* Allocate, append as buf fragement and construct the scan response
+	 * event.
+	 */
+	struct net_buf *scan_evt_buf = bt_buf_get_rx(BT_BUF_EVT, K_FOREVER);
+
+	net_buf_frag_add(buf, scan_evt_buf);
+
+	/* If scan data complete */
+	if (!scan_data_status) {
```

Comment:
The following code seem very similar to exiting code for handling advertising reports (which are, of course, also sent with the same HCI event as scan responses). Is it much effort to adjust the current code for advertising reports to reuse the same code for scan responses, as basically the only difference is the event type? (at least from a HCI point of view).

Reply:
Refactored in to ext_adv_info_fill()